### ISSUE TITLE
Tools for solidity server keeps crashing in WSL - windows

### ISSUE THREAD

**Issue by 0xfave** [Ackee-Blockchain/tools-for-solidity-vscode#66]
I have tried some of the commands [here](https://github.com/Ackee-Blockchain/tools-for-solidity-vscode/issues/44) but it's not working 

```py
Running 'woke lsp --port 43343'
[Error - 10:08:21 PM] Traceback (most recent call last):
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/server.py", line 268, in _task_done_callback
    task.result()
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 187, in run
    await self.__compilation_loop()
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 977, in __compilation_loop
    await self.__compile(self.__discovered_files)
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 848, in __compile
    self.__ir_reference_resolver.run_post_process_callbacks(
  File "/home/fave/.local/lib/python3.10/site-packages/woke/ast/ir/reference_resolver.py", line 174, in run_post_process_callbacks
    callback(callback_params)
  File "/home/fave/.local/lib/python3.10/site-packages/woke/ast/ir/meta/import_directive.py", line 165, in _post_process
    assert referenced_declaration is not None
AssertionError

[Info  - 10:08:21 PM] Connection to server got closed. Server will restart.
[Error - 10:09:11 PM] Traceback (most recent call last):
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/server.py", line 268, in _task_done_callback
    task.result()
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 187, in run
    await self.__compilation_loop()
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 977, in __compilation_loop
    await self.__compile(self.__discovered_files)
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 848, in __compile
    self.__ir_reference_resolver.run_post_process_callbacks(
  File "/home/fave/.local/lib/python3.10/site-packages/woke/ast/ir/reference_resolver.py", line 174, in run_post_process_callbacks
    callback(callback_params)
  File "/home/fave/.local/lib/python3.10/site-packages/woke/ast/ir/meta/import_directive.py", line 165, in _post_process
    assert referenced_declaration is not None
AssertionError

[Info  - 10:09:11 PM] Connection to server got closed. Server will restart.
[Error - 10:10:00 PM] Traceback (most recent call last):
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/server.py", line 268, in _task_done_callback
    task.result()
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 187, in run
    await self.__compilation_loop()
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 977, in __compilation_loop
    await self.__compile(self.__discovered_files)
  File "/home/fave/.local/lib/python3.10/site-packages/woke/lsp/lsp_compiler.py", line 848, in __compile
    self.__ir_reference_resolver.run_post_process_callbacks(
  File "/home/fave/.local/lib/python3.10/site-packages/woke/ast/ir/reference_resolver.py", line 174, in run_post_process_callbacks
    callback(callback_params)
  File "/home/fave/.local/lib/python3.10/site-packages/woke/ast/ir/meta/import_directive.py", line 165, in _post_process
    assert referenced_declaration is not None
AssertionError

[Info  - 10:10:00 PM] Connection to server got closed. Server will restart.
```

**Comment by 0xfave** [comment]
use this `woke lsp --port 54942` it says this

```py
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/fave/.local/bin/woke:8 in <module>                                                         │
│                                                                                                  │
│   5 from woke.cli.__main__ import main                                                           │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 8 │   sys.exit(main())                                                                         │
│   9                                                                                              │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/click/core.py:1157 in __call__                    │
│                                                                                                  │
│   1154 │                                                                                         │
│   1155 │   def __call__(self, *args: t.Any, **kwargs: t.Any) -> t.Any:                           │
│   1156 │   │   """Alias for :meth:`main`."""                                                     │
│ ❱ 1157 │   │   return self.main(*args, **kwargs)                                                 │
│   1158                                                                                           │
│   1159                                                                                           │
│   1160 class Command(BaseCommand):                                                               │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/rich_click/rich_group.py:21 in main               │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/click/core.py:1078 in main                        │
│                                                                                                  │
│   1075 │   │   try:                                                                              │
│   1076 │   │   │   try:                                                                          │
│   1077 │   │   │   │   with self.make_context(prog_name, args, **extra) as ctx:                  │
│ ❱ 1078 │   │   │   │   │   rv = self.invoke(ctx)                                                 │
│   1079 │   │   │   │   │   if not standalone_mode:                                               │
│   1080 │   │   │   │   │   │   return rv                                                         │
│   1081 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!                               │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/click/core.py:1688 in invoke                      │
│                                                                                                  │
│   1685 │   │   │   │   super().invoke(ctx)                                                       │
│   1686 │   │   │   │   sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)                    │
│   1687 │   │   │   │   with sub_ctx:                                                             │
│ ❱ 1688 │   │   │   │   │   return _process_result(sub_ctx.command.invoke(sub_ctx))               │
│   1689 │   │                                                                                     │
│   1690 │   │   # In chain mode we create the contexts step by step, but after the                │
│   1691 │   │   # base command has been invoked.  Because at that point we do not                 │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/click/core.py:1434 in invoke                      │
│                                                                                                  │
│   1431 │   │   │   echo(style(message, fg="red"), err=True)                                      │
│   1432 │   │                                                                                     │
│   1433 │   │   if self.callback is not None:                                                     │
│ ❱ 1434 │   │   │   return ctx.invoke(self.callback, **ctx.params)                                │
│   1435 │                                                                                         │
│   1436 │   def shell_complete(self, ctx: Context, incomplete: str) -> t.List["CompletionItem"]:  │
│   1437 │   │   """Return a list of completions for the incomplete value. Looks                   │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/click/core.py:783 in invoke                       │
│                                                                                                  │
│    780 │   │                                                                                     │
│    781 │   │   with augment_usage_errors(__self):                                                │
│    782 │   │   │   with ctx:                                                                     │
│ ❱  783 │   │   │   │   return __callback(*args, **kwargs)                                        │
│    784 │                                                                                         │
│    785 │   def forward(                                                                          │
│    786 │   │   __self, __cmd: "Command", *args: t.Any, **kwargs: t.Any  # noqa: B902             │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/click/decorators.py:33 in new_func                │
│                                                                                                  │
│    30 │   """                                                                                    │
│    31 │                                                                                          │
│    32 │   def new_func(*args: "P.args", **kwargs: "P.kwargs") -> "R":                            │
│ ❱  33 │   │   return f(get_current_context(), *args, **kwargs)                                   │
│    34 │                                                                                          │
│    35 │   return update_wrapper(new_func, f)                                                     │
│    36                                                                                            │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/woke/cli/lsp.py:53 in run_lsp                     │
│                                                                                                  │
│   50 │   config = WokeConfig()                                                                   │
│   51 │   config.load_configs()  # load ~/.woke/config.toml and ./woke.toml                       │
│   52 │                                                                                           │
│ ❱ 53 │   asyncio.run(run_server(config, port))                                                   │
│   54                                                                                             │
│                                                                                                  │
│ /usr/lib/python3.10/asyncio/runners.py:44 in run                                                 │
│                                                                                                  │
│   41 │   │   events.set_event_loop(loop)                                                         │
│   42 │   │   if debug is not None:                                                               │
│   43 │   │   │   loop.set_debug(debug)                                                           │
│ ❱ 44 │   │   return loop.run_until_complete(main)                                                │
│   45 │   finally:                                                                                │
│   46 │   │   try:                                                                                │
│   47 │   │   │   _cancel_all_tasks(loop)                                                         │
│                                                                                                  │
│ /usr/lib/python3.10/asyncio/base_events.py:649 in run_until_complete                             │
│                                                                                                  │
│    646 │   │   if not future.done():                                                             │
│    647 │   │   │   raise RuntimeError('Event loop stopped before Future completed.')             │
│    648 │   │                                                                                     │
│ ❱  649 │   │   return future.result()                                                            │
│    650 │                                                                                         │
│    651 │   def stop(self):                                                                       │
│    652 │   │   """Stop running the event loop.                                                   │
│                                                                                                  │
│ /home/fave/.local/lib/python3.10/site-packages/woke/cli/lsp.py:28 in run_server                  │
│                                                                                                  │
│   25 │   │   writer.close()                                                                      │
│   26 │   │   logger.info("Client disconnected")                                                  │
│   27 │                                                                                           │
│ ❱ 28 │   server = await asyncio.start_server(client_callback, port=port)                         │
│   29 │   logger.info(f"Started LSP server on port {port}")                                       │
│   30 │                                                                                           │
│   31 │   async with server:                                                                      │
│                                                                                                  │
│ /usr/lib/python3.10/asyncio/streams.py:85 in start_server                                        │
│                                                                                                  │
│    82 │   │   │   │   │   │   │   │   │   │   loop=loop)                                         │
│    83 │   │   return protocol                                                                    │
│    84 │                                                                                          │
│ ❱  85 │   return await loop.create_server(factory, host, port, **kwds)                           │
│    86                                                                                            │
│    87                                                                                            │
│    88 if hasattr(socket, 'AF_UNIX'):                                                             │
│                                                                                                  │
│ /usr/lib/python3.10/asyncio/base_events.py:1519 in create_server                                 │
│                                                                                                  │
│   1516 │   │   │   │   │   try:                                                                  │
│   1517 │   │   │   │   │   │   sock.bind(sa)                                                     │
│   1518 │   │   │   │   │   except OSError as err:                                                │
│ ❱ 1519 │   │   │   │   │   │   raise OSError(err.errno, 'error while attempting '                │
│   1520 │   │   │   │   │   │   │   │   │     'to bind on address %r: %s'                         │
│   1521 │   │   │   │   │   │   │   │   │     % (sa, err.strerror.lower())) from None             │
│   1522 │   │   │   │   completed = True                                                          │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
OSError: [Errno 98] error while attempting to bind on address ('0.0.0.0', 54942): address already in use
```

**Comment by michprev** [comment]
Hey @0xfave,
the assertion error should be fixed in Tools for Solidity 1.10.4. Regarding the second error, it seems that port 54942 was already being used on your device.

**Comment by 0xfave** [comment]
Thank you
